### PR TITLE
Fix TORCH_HOME environment variable in docs

### PR DESCRIPTION
**Pull request by stephengmatthews**
Replaced `TORCH_MODEL_ZOO`, which is deprecated, with `TORCH_HOME` in docs.

`TORCH_MODEL_ZOO` is not used so `TORCH_HOME` is the only environment variable to set the cache directory.

### CODE REVIEW

File: docs/source/models.rst [hunk 1]
```diff
@@ -127,7 +127,7 @@ These can be constructed by passing ``pretrained=True``:
     regnet_x_32gf = models.regnet_x_32gf(pretrained=True)
 
 Instancing a pre-trained model will download its weights to a cache directory.
-This directory can be set using the `TORCH_MODEL_ZOO` environment variable. See
+This directory can be set using the `TORCH_HOME` environment variable. See
 :func:`torch.utils.model_zoo.load_url` for details.
```
Comment (NicolasHug) [inline docs/source/models.rst:131]: This is actually outdated as well:

```suggestion
:func:`torch.hub.load_state_dict_from_url` for details.
```

**Review by NicolasHug — APPROVED** [review]
Thanks @stephengmatthews ! This will partially address https://github.com/pytorch/vision/issues/4795

One minor suggestion below, but LGTM when adressed. Thanks!

**Comment by stephengmatthews** [comment]
> Thanks @stephengmatthews ! This will partially address #4795
> 
> One minor suggestion below, but LGTM when adressed. Thanks!

Thanks @NicolasHug , I hadn't spotted the open issue! I've made the suggested change, too.

**Comment by NicolasHug** [comment]
Thanks @stephengmatthews ,

We're unable to merge for now because of the CLA - would you mind taking a look at https://github.com/pytorch/vision/pull/4843#issuecomment-958767462 ? I think it can take time to be processed so if you've already done it, nothing more to do on your end :)

**Comment by stephengmatthews** [comment]
> Thanks @stephengmatthews ,
> 
> We're unable to merge for now because of the CLA - would you mind taking a look at [#4843 (comment)](https://github.com/pytorch/vision/pull/4843#issuecomment-958767462) ? I think it can take time to be processed so if you've already done it, nothing more to do on your end :)

Signed it, so guess it's a waiting game now, thanks again.

**Pull request merged**